src/shared/istiopolicy/creator.go

```diff
@@ -140,6 +148,33 @@ func (c *Creator) saveServiceAccountName(ctx context.Context, clientIntents *v1a
 	return nil
 }
 
+func (c *Creator) saveSideCarStatus(ctx context.Context, clientIntents *v1alpha2.ClientIntents, missingSideCar bool) error {
+	oldValue, ok := clientIntents.Annotations[v1alpha2.OtterizeMissingSidecarAnnotation]
+	if ok && oldValue == strconv.FormatBool(missingSideCar) {
+		return nil
+	}
+
+	updatedIntents := clientIntents.DeepCopy()
+	if updatedIntents.Annotations == nil {
+		updatedIntents.Annotations = make(map[string]string)
+	}
+
+	updatedIntents.Annotations[v1alpha2.OtterizeMissingSidecarAnnotation] = strconv.FormatBool(missingSideCar)
+	err := c.client.Patch(ctx, updatedIntents, client.MergeFrom(clientIntents))
+	if err != nil {
+		logrus.WithError(err).Errorln("Failed updating intent with sidecar status")
+		c.recorder.RecordWarningEventf(clientIntents, ReasonIntentsLabelFailed, "Failed updating intent with service account name: %s", err.Error())
+		return err
+	}
+
+	if missingSideCar {
+		logrus.Infof("updating intent %s with side %v", clientIntents.Name, missingSideCar)
```

Comment:
Probably no need for the log as you are already sending it as an event on the resource, and the user will be able to see it there. IMO it's better to keep the operator log cleaner so that the user will have an easier time seeing errors that require attention, if there are any.